utils/terrascript_client.py

```diff
@@ -1726,3 +1735,183 @@ def unpack_namespace_info(self, namespace_info):
         cluster = namespace_info['cluster']['name']
         namespace = namespace_info['name']
         return cluster, namespace
+
+    @staticmethod
+    def validate_elasticsearch_version(version):
+        """ Validate ElasticSearch version. """
+        return version in [7.4, 7.1,
+                           6.8, 6.7, 6.5, 6.4, 6.3, 6.2, 6.0,
+                           5.6, 5.5, 5.3, 5.1,
+                           2.3,
+                           1.5]
+
+    @staticmethod
+    def get_elasticsearch_service_role_tf_resource(identifier):
+        """ Service role for ElasticSearch. """
+        service_role = {
+          'aws_service_name': "es.amazonaws.com"
+        }
+        return aws_iam_service_linked_role(identifier, **service_role)
+
+    @staticmethod
+    def is_elasticsearch_domain_name_valid(name):
+        """ Handle for Error creating Elasticsearch:
+        InvalidParameterValue: Elasticsearch domain name must must start with a
+        lowercase letter and must be between 3 and 28 characters. Valid
+        characters are a-z (lowercase only), 0-9, and - (hyphen). """
+        if len(name) < 3 or len(name) > 28:
+            return False
+        pattern = r'^[a-z][a-z0-9-]+$'
+        return re.search(pattern, name)
+
+    def populate_tf_resource_elasticsearch(self, resource, namespace_info,
+                                           existing_secrets):
+
+        account, identifier, values, output_prefix, output_resource_name = \
+            self.init_values(resource, namespace_info)
+
+        tf_resources = []
+
+        self.init_common_outputs(tf_resources, namespace_info,
+                                 output_prefix, output_resource_name)
+
+        if values['identifier'] and \
+           not self.is_elasticsearch_domain_name_valid(values['identifier']):
+            raise ElasticSearchResourceNameInvalidError(
+                f"[{account}] ElasticSearch domain name must must start with" +
+                f" a lowercase letter and must be between 3 and 28 " +
+                f"characters. Valid characters are a-z (lowercase only), 0-9" +
+                f", and - (hyphen). " +
+                f"{values['identifier']}")
+
+        # values['domain_name'] = identifier
+        elasticsearch_version = values.get('elasticsearch_version')
+        if not self.validate_elasticsearch_version(elasticsearch_version):
+            raise ElasticSearchResourceVersionInvalidError(
+                f"[{account}] Invalid ElasticSearch version" +
+                f" {values['elasticsearch_version']} provided" +
+                f" for resource {values['identifier']}.")
+
+        values['cluster_config']["zone_awareness_enabled"] = False
+
+        es_values = {}
+        es_values["domain_name"] = identifier
+        es_values["elasticsearch_version"] = \
+            values.pop('elasticsearch_version')
```

Comment:
Why pop?  Why not use the variable you set/checked earlier?

Reply:
good catch